### DISCUSSION TITLE
CLN --experimental-offers for BOLT12 Offer / OCEAN Mining

### DISCUSSION THREAD

**Discussion by neversleep404** [raspiblitz/raspiblitz#4638] [Feature / App Ideas]
https://ocean.xyz/docs/lightning

OCEAN uses BOLT12 offers for Lightning payouts.

To ensure the security and anonymity of the payouts, OCEAN requires that a message is signed linking your OCEAN bitcoin address to a BOLT12 offer.

To do this, message signing will need to be supported by your wallet and/or signing device.

Once you have generated a BOLT12 offer, OCEAN will provide you with a message to sign.

**Comment by rootzoll** [comment]
Can you sepcify what you need from RaspiBlitz? 

If just need to sign a message with your Core Lighting node have you tried `lightning-cli signmessage MESSAGE_TO_SIGN`?

**Reply by neversleep404** [reply]
Ocean.xyz Note: to do this you will need to launch Core Lightning with the flag --experimental-offers

**Reply by neversleep404** [reply]
admin@192.168.178.99:~ ₿ lightning-cli offer any bc1xxxxxxx
{
"code": -1,
"message": "experimental-offers not enabled"
}
admin@192.168.178.99:~ ₿

**Comment by neversleep404** [comment]
Solution:
Using the System Menue and insert **### experimental-offers** in the config
